src/valid/type.rs

```diff
@@ -332,7 +335,12 @@ impl super::Validator {
                         return Err(TypeError::InvalidData(member.ty));
                     }
                     if top_level && !base_info.flags.contains(TypeFlags::INTERFACE) {
-                        return Err(TypeError::InvalidBlockType(member.ty));
+                        // Certain non-`INTERFACE` types like arrays and `bool` are
+                        // permitted for built-ins, whose types we verify in
+                        // `VaryingContext::validate`.
+                        if !matches!(member.binding, Some(crate::Binding::BuiltIn(_))) {
```

Comment:
let's not use matches please

Reply:
Here, `matches!` doesn't buy much, but there's another spot I ran into (in another patch series) where the `match` is worse. I'll point it out when we get there, curious what you'll think.